Update wine.owl

bvarner-ebi opened the pull request:
Seeking feedback on the following:

Any thoughts on the addition of suffixes here (_color, _varietal, _wine)?

Should subclasses of **color** be changed to instances of **color**? If yes, then I would remove the _color suffixes.

Should capitalisation rules be defined / recorded somewhere?

Is there a standard on using / avoiding accented letters (e.g., é?) I put the in the annotations for rose (rosé).

dosumis commented:
Looks good

> Seeking feedback on the following:
> 
> Any thoughts on the addition of suffixes here (_color, _varietal, _wine)?

I think they are useful disambiguation.  In a  but in the context of an application, shorter names would probably be better
 
> Should subclasses of color be changed to instances of color? If yes, then I would remove the _color suffixes.

I think classes make sense here. Someone might reasonably want to extend with subclasses, e.g. by adding light or dark red.  OTOH, perhaps the names are more conventional modifiers to described classes of wine - not tracking colour too closely. 
 
> Should capitalisation rules be defined / recorded somewhere?

For OBO ontologies, the convention is not to capitalise except in the case of proper names.  I wasn't paying too much

> Is there a standard on using / avoiding accented letters (e.g., é?) I put these in the annotations for rose (rosé).

Include accents, but important to ensure they are UTF-8.

dosumis commented:
The specific Rose wines could be auto-classified using inference.  Can you edit to do this?

In this case it may seem trivial, but in general I think it is useful to keep classification dynamic

bvarner-ebi commented:
Based on @dosumis's feedback, I updated the commit:

- removed _color suffix and will keep color subclasses (will not make them instances)

- kept _varietal and _wine suffixes since they could be confused due to similar names between the two subclasses (still open to removing based on additional feedback)

- capitalisation seems to be following proper noun capitalisation only (no change)

- accented letters used for subclass names, accented letters noted in annotations (no change)

- changed specific rose wines to auto-classify

Kindly advise if additional changes are needed; otherwise, please merge with the base branch.

bvarner-ebi commented:
**Wine ontology exercises**
  
Write DL queries for:
wines that are made from Nebbiolo verietal grapes

**BV: wine that made_from some Nebbiolo_varietal**

wines from Piedmont
		
**BV: wine that grown_in value Piedmont**

Given that 'Chardonnay wine' is a generic names for wines made from Chardonnay (varietal) grapes, add a term for Chardonnay (wine) with an appropriate logical definition. What does the reasoner classify as Chardonnay?

**BV: white_wine**

View the relationships (subClassOf axioms) of Chianti (wine). How might these frustrate attempts to classify wines using the pattern you just used to define Chardonnay wine?

**BV: Chianti has multiple varietals**

Add a term for Rosé wine or orange wine and some terms for specific wines that will be auto-classified under it.

**More advanced exercises**
Add axiomatization to detect an error (inconsistency) if a user adds a wine that is asserted or inferred to be both red and white.
	
**BV: add disjoint axiom to colour classes; after running the reasoner, this would generate an 	‘owl:Nothing’ result to a wine inferred to be more than one colour**

Following the patterns used in the ontology, add a region of Germany and a wine from that region and a general class for German wine.
	
Explain, in clear English, why the reasoner classifies this wine as German.
	
**BV: Because I asserted that German wine is grown in Germany, Goldiesling wine is grown in Saxony and Saxony is a region of Germany, the reasoner deduced that Goldriesling wine is German wine. (Thoughts… maybe the grown_in object property should be related to the varietals OR the object property should be renamed to’ produced_in’ since the wine (if conceptualised as the imbibed form) is not “grown”, but rather produced, from the varietals grown in the regions.**


Add some individual wines following the pattern used to define Barolo_Villero_2015
	
**BV: ++ La_Chablisienne_Chablis_Chateau_Grenouilles_2015 and 						Schloss_Wackerbarth_Goldriesling_trocken_2009**


Extend the ontology to add terms for tasting notes and use these to classify Individual wines?

**BV: I did this, but get an error in the log file after running the synchronizer.**
	
_“WARN  10:26:32  Reasoning may be incomplete! See log level INFO for more details.
   	INFO  10:26:33  ELK does not support querying equivalent classes and subclasses of 		ObjectComplementOf.”_

	**Is this an issue?**

dosumis commented:
Looks great. The axiomatisation all makes sense as results in correct classifications. Your decision to treat tasting notes as individuals makes sense I think, as it is probably challenging to make a classification from these terms.  However. If you wanted to model tasting notes as classes, you could do this by adding type axioms to individuals, 

e.g. 
Type: has_tasting_note some grassy

One other issue:

> Add axiomatization to detect an error (inconsistency) if a user adds a wine that is asserted or inferred to be both red and white.

> BV: add disjoint axiom to colour classes; after running the reasoner, this would generate an ‘owl:Nothing’ result to a wine asserted or inferred to be more than one colour

In this case you would only get an inconsistency if someone added this restriction to a wine:

has_color some (red and white)

But this seems an unlikely error.  There are a couple of other approaches you could use that would catch cases where a wine ends up with the restrictions:

has_color some red
has_color some white

1. Make 'red wine' and 'white wine' disjoint
2. Use a general class axiom to say that no entity has_color (some) red and has_color (some) white
![image](https://user-images.githubusercontent.com/112839/146176628-b3d71d45-2f78-40a3-95f5-6aee2d76a91b.png)

dosumis commented:
> Extend the ontology to add terms for tasting notes and use these to classify Individual wines?

> BV: I did this, but get an error in the log file after running the synchronizer.

> “WARN 10:26:32 Reasoning may be incomplete! See log level INFO for more details.
INFO 10:26:33 ELK does not support querying equivalent classes and subclasses of ObjectComplementOf.”

> Is this an issue?

This is warning of some limitations of the ELK reasoner.  I'm not sure what has triggered this though, and the classification of individual wines by tasting notes is working.

bvarner-ebi commented:
Thanks for the feedback, @dosumis.
I removed the disjoint between the wine colors and added a disjoint between red_wine and white_wine.
